repos/system_upgrade/el9toel10/actors/mysqlcheck/libraries/mysqlcheck.py

```diff
@@ -0,0 +1,123 @@
+from leapp import reporting
+from leapp.libraries.common.rpms import has_package
+from leapp.libraries.stdlib import api
+from leapp.models import DistributionSignedRPM
+
+import subprocess
+
+
+class MySQLCheckLib:
+    REMOVED_ARGS = [
+        '--avoid-temporal-upgrade',
+        '--show-old-temporals',
+        '--old',
+        '--new',
+        '--default-authentication-plugin',
+        '--no-dd-upgrade',
+        '--language',
+        '--ssl',
+        '--admin-ssl',
+        '--character-set-client-handshake',
+        '--old-style-user-limits',
+    ]
+
+    # Summary for mysql-server report
+    report_server_inst_summary = (
+        'MySQL server component will be upgraded.\n'
+        'Since RHEL-10 includes MySQL server 8.4 by default, it might be necessary'
+        ' to proceed with additional steps.'
+        '\n'
+    )
+
+    report_server_inst_hint = (
+        'Back up your data before proceeding with the upgrade'
+        ' and follow steps in the documentation section "Migrating to a RHEL 10 version of MySQL"'
+        ' after the upgrade.'
+    )
+
+    # Link URL for mysql-server report
+    report_server_inst_link_url = 'https://access.redhat.com/articles/7099234'
+
+    # Default title
+    report_title = 'MySQL (mysql-server) has been detected on your system'
+
+    # Default severity
+    report_severity = reporting.Severity.MEDIUM
+
+    found_arguments = set()
+    found_options = set()
+
+    def _generate_report(self):
+        """
+        Create report on mysql-server package installation detection.
+
+        Should remind user about present MySQL server package
+        installation, warn them about necessary additional steps, and
+        redirect them to online documentation for the upgrade process.
+        """
+
+        if self.found_arguments or self.found_options:
+            self.report_severity = reporting.Severity.HIGH
+            self.report_title = 'MySQL (mysql-server) seems to be using deprecated config options'
+            self.report_server_inst_summary += (
+                '\nWarning:\n'
+                'It seems that some config options currently used for MySQL server'
+                ' will be removed in updated MySQL server.\n'
+                'If you proceed with the update now, without addressing this issue,'
+                ' the MySQL server will fail to start until the config is fixed.\n'
+                'Detected deprecated options:\n'
+            )
+            for arg in self.found_options:
+                self.report_server_inst_summary += f"{arg} in MySQL config file\n"
+            for arg in self.found_arguments:
+                self.report_server_inst_summary += f"{arg} in SystemD service override\n"
```

Comment:
This will lead to "lot of" (depending on number of possible entries) text redundancy which is not accepted well by customers. I expect this will not generate usually too many lines, however, it's better to introduce the meaning of the list and then just simply list all the stuff. Something like:
```
Following configured MySQL options will are not valid anymore for MySQL on the target system:
    - option1
    - option2
    - ...
Following systemd...:
    - ...
    - ...
```

Also, define please the `FMT_LIST_SEPARATOR = '\n    - '` and use it for the listings so it's consistent between actors. We plan to document this and most likely create a function in the shared library - refactoring all actors to use it. But let's go with the simple formatting now.